File: data/readable_format/fortran/file2175.f90
PROGRAM TestEquivalence
  IMPLICIT NONE
  REAL :: A
  REAL :: B
  REAL :: expectedValue

  EQUIVALENCE(A, B)

  A = 0.0
  expectedValue = 0.0
  CALL Verify(A, B, expectedValue)

  B = 5.0
  expectedValue = 5.0
  CALL Verify(A, B, expectedValue)

CONTAINS

  SUBROUTINE Verify(x, y, expected)
    REAL, INTENT(IN) :: x, y, expected
    IF (x == expected .AND. y == expected) THEN
      PRINT *, "Test Passed: A =", x, ", B =", y, ", Expected =", expected
    ELSE
      PRINT *, "Test Failed: A =", x, ", B =", y, ", Expected =", expected
    END IF
  END SUBROUTINE Verify

END PROGRAM TestEquivalence

File: data/readable_format/cpp/file2175.cpp
#include <iostream>
#include <cassert>

// Using union for equivalence as in the original translation
union Equivalence {
    float A;
    float B;
};

void verify(float x, float y, float expected) {
    if (x == expected && y == expected) {
        std::cout << "Test Passed: A = " << x << " B = " << y << " Expected = " << expected << std::endl;
    } else {
        std::cerr << "Test Failed: A = " << x << " B = " << y << " Expected = " << expected << std::endl;
        std::abort(); // Use assert or abort to halt if this was a real unit test.
    }
}

int main() {
    Equivalence eq;

    eq.A = 0.0f; // Initialize A
    verify(eq.A, eq.B, 0.0f);

    eq.B = 5.0f; // Modify B, which also modifies A due to union
    verify(eq.A, eq.B, 5.0f);

    return 0;
}